app/javascript/gobierto_data/webapp/pages/Home.vue

```diff
@@ -1,24 +1,250 @@
+  },
+  mixins: [categoriesMixin],
+  props: {
+    currentComponent: {
+      type: String,
+      required: true,
+      default: ''
+    },
+    activateTabSidebar: {
+      type: Number,
+      default: 0
+    }
   },
   data() {
     return {
-      activeTabIndex: 0
+      items: store.state.items || [],
+      subsetItems: [],
+      currentView: '',
+      currentTab: 0,
+      filters: store.state.filters || [],
+      activeFilters: store.state.activeFilters || new Map(),
+      defaultFilters: store.state.defaultFilters || new Map(),
+      isFetchingData: false
+    }
+  },
+  async created() {
+    this.currentView = this.currentComponent
+    if (this.$route.params.tabSidebar === 0) {
+      this.currentTab = this.$route.params.tabSidebar
+    } else {
+      this.currentTab = this.activateTabSidebar
+    }
+    this.$root.$on("sendCheckbox", this.handleCheckboxStatus)
+    this.$root.$on("selectAll", this.handleIsEverythingChecked)
+    if (this.items.length) {
+      this.updateDOM();
+    } else {
+
+      const { items, filters } = await this.getItems();
+
+      this.isFetchingData = true;
+
+      this.items = items;
+
+      this.defaultFilters = filters
+      this.filters = filters;
+
+      this.updateDOM();
+    }
+    this.isFetchingData = true;
+  },
+  methods: {
+    setCurrentView() {
+      this.currentView = 'DataSets'
+    },
+    async getItems() {
+      const [
+        {
+          data: { data: __items__ = [] }
+        },
+        {
+          data: {
+            data: attributesDictionary = [],
+            meta: filtersFromConfiguration
+          }
+        }
+      ] = await axios.all([
+        axios.get(`${baseUrl}/datasets`),
+        axios.get(`${baseUrl}/datasets/meta?stats=true`)
+      ]);
+
+      const { availableFilters } = CONFIGURATION
+      // Middleware receives both the dictionary of all possible attributes, and the selected filters for the site
+      this.middleware = new Middleware({
+        dictionary: attributesDictionary,
+        filters: availableFilters
+      });
+      store.addDatasets(__items__);
+      let items = this.setData(__items__);
+      let filters = [];
+
+      if (filtersFromConfiguration) {
+
+        items = items.map(item => ({ ...item }));
+
+        filters = this.middleware.getFilters(filtersFromConfiguration) || [];
+
+        if (filters.length) {
+          this.activeFilters = new Map();
+
+          // initialize active filters
+          filters.forEach(filter =>
+            this.activeFilters.set(filter.key, undefined)
+          );
+
+          // save the filters
+          store.addFilters(filters);
+          store.addDefaultFilters(filters);
+        }
+      }
+
+      // Assign this object BEFORE next function for better performance
+      this.subsetItems = items;
+
+
+      // Optional callback to update data in background, setup in CONFIGURATION object
+      const itemsUpdated = await this.alterDataObjectOptional(items);
```

Comment:
You've full-copied this file. There are useless things for `gobierto-data`, aren't there? For instance, this method I know is no longer there.